File: java/pl/edu/pg/mygame/organisms/plants/Belladonna.java
package pl.edu.pg.mygame.organisms.plants;

import pl.edu.pg.mygame.Grid;
import pl.edu.pg.mygame.World;
import pl.edu.pg.mygame.organisms.Organism;
import pl.edu.pg.mygame.organisms.Species;

import java.awt.*;

public class Belladonna extends Plant {
    public Belladonna(Point p) {
        super(p);
        this.species = Species.BELLADONNA;
        this.strength = 99;
    }

    @Override
    public boolean collision(Organism other, boolean attacking, Grid grid) {
        World.message_panel.showMessage("Belladonna killed " + other.getSpecies() + " at: " +
                String.format("%d,%d)\n", other.getPosition().x, other.getPosition().y));
        grid.setOrganism(other.getPosition(), null);
        return true;
    }
}

File: python/organisms/plants/belladonna.py
from gui.logger import Logger
from organisms.plants.plant import Plant
from organisms.species import Species


class Belladonna(Plant):
    def __init__(self, pos):
        super().__init__(pos)
        self.strength = 99
        self.species = Species.BELLADONNA

    def collision(self, other, attacking, grid):
        Logger.log('Belladonna killed ' + other.species.name + ' at: ' + str(other.position))
        grid.set_organism(other.position, None)
        return True
